### ISSUE TITLE
[New Primitive] Autocomplete Popup

### ISSUE THREAD

**Issue by ccorcos** [radix-ui/primitives#2337]
### Overview

When typing into an input, its common to have a popup similar to a dropdown menu that is triggered without clicking a button.

Examples:
- Filling out an address and searching completions with google
- Searching for users in an app
- `:emoji:` menu on GitHub (not an input, but a textarea with an arbitrary trigger)

I think it would be easiest if the DropdownMenu.Trigger could be manually opened or closed and passed a ref that isn't the Trigger button.

Also, think it would be great to publish whatever component does the layout of aligning the popup menu. It was really hard to follow where that happens in the code, but I do see the triggerRef getting tied to the context. This just seems like a valuable component to have around in case I wanted to build this autocomplete thing myself.

### Examples in other libraries

I don't know many UI libraries.
- https://material.angular.io/components/autocomplete/overview

**Comment by joaom00** [comment]
Hey @ccorcos, thank you for the suggestion. Already have a feature request for that primitive #1342

**Comment by ccorcos** [comment]
Yeah, I saw that. But the difference is (1) the search input doesn’t need
to be inside the pop up and (2) this isn’t a multi select.

It seems like the Combobox primitive is going to be pretty specific… no?

**Comment by ccorcos** [comment]
I guess this is called a combobox 😄
